config/formats/typescriptExportDefinition.ts

```diff
@@ -32,53 +33,59 @@ const getTokenType = (tokenTypesPath: string): string => {
     return designTokenType
   } catch (e) {
     // eslint-disable-next-line no-console
-    console.error(`Error trying to load design token type from file "${tokenTypesPath}".`)
+    throw new Error(`Error trying to load design token type from file "${tokenTypesPath}".`)
   }
-  return ''
 }
 
+/**
+ * @description throws an error stating the token name, type and invalid value
+ * @param token w3cTransformedToken
+ */
+const invalidTokenValueError = (token: w3cTransformedToken) => {
+  throw new Error(`Invalid token: "${token.name}" with type "${token.$type}" can not have a value of "${token.value}"`)
+}
 /**
  * convertPropToType
  * @description return take type attribute and return correct typescript type as string
  * @param type
  * @returns
  */
-const convertPropToType = (value: unknown, type?: string): string => {
-  switch (type) {
+const convertPropToType = (token: w3cTransformedToken): string => {
+  if (!Object.prototype.hasOwnProperty.call(token, 'value')) {
```

Comment:
Have you considered using `hasOwn` instead here? It's easier to read and mitigates object prototype inheritance problems. [Link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn).


```suggestion
  if (!Object.hasOwn(token, 'value')) {
```

Reply:
Yes, I had some ts issues some time ago, but I will give it another try

Reply:
I get this:
<img width="1608" alt="Screenshot 2022-11-22 at 12 27 08" src="https://user-images.githubusercontent.com/813754/203302900-dd5b3604-04e1-49db-a845-5a5da2a1b629.png">

Even though my `tsconfig.json` has this:
```json
    "lib": ["ES2019.Array", "ES2022"],
  ```

Reply:
oh weird, should be in the ES2022 spec.. not sure what issue without getting hands-on with it, so can leave as-is for now